src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -4685,6 +4682,26 @@ ITermDispatch::StringHandler AdaptDispatch::_RestoreTabStops()
     };
 }
 
+void AdaptDispatch::_ReturnCsiResponse(const std::wstring_view response) const
+{
+    const auto csi = _terminalInput.GetInputMode(TerminalInput::Mode::SendC1) ? L"\x9B" : L"\x1B[";
+    _api.ReturnResponse(fmt::format(FMT_COMPILE(L"{}{}"), csi, response));
+}
+
+void AdaptDispatch::_ReturnDcsResponse(const std::wstring_view response) const
+{
+    const auto dcs = _terminalInput.GetInputMode(TerminalInput::Mode::SendC1) ? L"\x90" : L"\x1BP";
+    const auto st = _terminalInput.GetInputMode(TerminalInput::Mode::SendC1) ? L"\x9C" : L"\x1B\\";
+    _api.ReturnResponse(fmt::format(FMT_COMPILE(L"{}{}{}"), dcs, response, st));
+}
+
+void AdaptDispatch::_ReturnOscResponse(const std::wstring_view response) const
```

Comment:
Thought: Xterm specifies that OSC responses are returned with the same terminator (`ST` or `BEL`) as the request that generated them. I suppose having this function affords us a much easier way to find/fix all occurrences of us being out of spec :)

>         XTerm accepts either BEL  or ST  for terminating OSC
>         sequences, and when returning information, uses the same
>         terminator used in a query.  While the latter is preferred,
>         the former is supported for legacy applications:
>         o   Although documented in the changes for X.V10R4 (December
>             1986), BEL  as a string terminator dates from X11R4
>             (December 1989).
>         o   Since XFree86-3.1.2Ee (August 1996), xterm has accepted ST
>             (the documented string terminator in ECMA-48).

Reply:
I really hate that `BEL` terminator, but you're right, we should be responding with `BEL` if the query was terminated with `BEL`.